src/async_fuse/fuse/fuse_reply.rs

```diff
@@ -160,59 +164,51 @@ impl ReplyRaw {
         Self { unique, fd }
     }
 
-    /// Send response to FUSE kernel
-    async fn send(self, data: impl AsIoVecList + Send + Sync + 'static) -> nix::Result<usize> {
+    /// Send raw message to FUSE kernel
+    async fn send_raw_message(
+        self,
+        error: i32,
+        data: impl AsIoVecList + Send + Sync + 'static,
+    ) -> nix::Result<usize> {
         let fd = self.fd;
         let wsize = smol::unblock(move || {
             let header_len = mem::size_of::<FuseOutHeader>();
 
             let header = FuseOutHeader {
                 len: (header_len.overflow_add(data.len())).cast(),
-                error: 0, // FUSE requires the error number to be negative
+                error,
                 unique: self.unique,
             };
 
             let header_bytes = abi_marker::as_abi_bytes(&header);
 
-            let (single, mut vecs);
-            let iovecs: &[IoVec<&[u8]>] = if data.len() > 0 {
-                vecs = data.as_io_vec_list();
-                vecs.insert(0, IoVec::from_slice(header_bytes));
-                vecs.as_slice()
-            } else {
-                single = [IoVec::from_slice(header_bytes)];
-                &single
-            };
+            let mut iovecs = vec![IoVec::from_slice(header_bytes)];
```

Comment:
This operation will copy the whole slice, do we really need to copy it?

Reply:
Done